### PR TITLE
[LTR] Update the feature name from "learn to rank" to "learning to rank".

### PR DESCRIPTION
**Pull request by afoucret**
Just a simple s/learning/learn on the Learn to rank code as requested by @joshdevins

### CODE REVIEW

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  */
 public enum FeatureFlag {
     TIME_SERIES_MODE("es.index_mode_feature_flag_registered=true", Version.fromString("8.0.0"), null),
-    LEARN_TO_RANK("es.learn_to_rank_feature_flag_enabled=true", Version.fromString("8.10.0"), null),
+    LEARNING_TO_RANK("es.learning_to_rank_feature_flag_enabled=true", Version.fromString("8.10.0"), null),
```
Comment (kderusso) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java:19]: Didn't catch this in the previous PR, but should we update this to version `8.12.0`?

**Review by kderusso — APPROVED** [review]
Thank you for doing a large scale refactor/rename in a separate PR 👍

**Review by demjened — APPROVED** [review]
LookING Good To Me!

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java (hunk 1 above)
Comment (afoucret) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java:19]: Good catch. I did fix the version here and in 2 other places.
Thank you!!!

**Pull request merged**